source/language/delays.rst

```diff
@@ -125,14 +125,17 @@ these side effects. Also contrary to TeX, we prohibit overlapping gates.
 Operations on durations
 -----------------------
 
-We can add/subtract two durations, or multiply them by a constant, to get new
-duration. The result must be positive. These are compile time operations since ultimately all
+We can add/subtract two durations, or multiply or divide them by a constant, to get a new
+duration. Division of two durations results in a machine-precision float 
+(see :ref:`divideDuration`). Negative durations are allowed, however
+passing a negative duration to a ``gate[duration]`` or ``box[duration]`` expression will result in an error.
+All operations on durations happen at compile time operations since ultimately all
```

Comment:
I think everything in this PR looks fine to me except the duration part. I have some questions:
- Is there a need for casting durations to floats? What is an example?
- What does it even mean for durations to be cast to floats? durations have units, floats do not. Dropping the units makes the conversion semantics non-preserving.
- Even if there is a need for casting, do we need duration here? I think the "multiply by constant" in the original spec should cover that already since you can e.g. do `.5 * duration`

Reply:
There is no cast of duration to float for the reasons you gave.  You can do the binary operation `duration / duration -> float` though, which doesn't have the unit problem, because it dimensionally reduces to a scalar, rather than something with dimensions of time.  That's for cases like manual dynamic decoupling, where say you want to do
```
n_reps = dur1 / dur2;
long_operation[dur1]($0, $1);
for i in [1:n_reps] {
    dynamic_decouple[dur2]($2);
}
```

(this section isn't on the same page as the casting - it's in the bit about delays)

Reply:
ok i think i understand now. being able to compute the ratio of two durations sounds good to me.